irony.el

```diff
@@ -661,6 +661,17 @@ If no such file exists on the filesystem the special file '-' is
                              (format "%s\n"
                                      (combine-and-quote-strings argv)))))))
 
+(defun irony--send-request-sync (request callback &rest args)
```

Comment:
My thought was that once a synchronous request has been made, all previous requests must finish. I guess that I missed that callbacks can themselves register new requests? Or are you thinking about something else?